utils/utils.go

```diff
@@ -33,11 +33,23 @@ func ParseArgs() cliArgs {
 	}
 }
 
-func RemoveIndexFromSlice(slice []string, idx int) ([]string, error) {
-
-	if idx < 0 || idx >= len(slice) {
-		return slice, errors.New("index out of range")
+func Complement[T comparable](slice, toRemove []T) []T {
+	var complement []T
+
+	for _, element := range slice {
+		found := false
+		for _, remove := range toRemove {
+			if element == remove {
+				fmt.Printf("\nExcluding: %v", element)
+				found = true
+				break
+			}
+		}
+		if !found {
+			complement = append(complement, element)
+		}
```

Comment:
I think the time complexity of this algorithm can be improved, at the moment it's O(n*m) because in the worst case scenario if both slice and toRemove have 50 elements we're looking at 250 iterations. Not a likely scenario, but still!

This could easily be improved with hash sets, but TIL Go doesn't have sets in its standard library (🤡) so it looks like you're going to have to implement set semantics yourself.
It should be pretty straightforward:

1. Create a hash map
2. Insert all the elements from `toRemove` into the map as keys with empty values
3. Iterate over `slice` and check whether an element is present in the hash map. If it isn't add it to `complement`
4. Return `complement`

That way we avoid the nested loop completely, and we only iterate over each slice only once (:

Reply:
Good shout! You can tell I did not study computer science 😁